lib/evss/letters/service.rb

```diff
@@ -42,6 +48,15 @@ def handle_error(error)
           super(error)
         end
       end
+
+      def log_edipi
+        InvalidLetterAddressEdipi.find_or_create_by(edipi: @user.edipi)
+      end
+
+      def invalid_address_error?(error)
+        return false unless error.is_a?(Common::Client::Errors::ClientError) && error.body.is_a?(Hash)
```

Comment:
I'm assuming this escape clause is in case `error.body` == `nil`? I think the usage of safe operators would do the same job since `nil` evaluates falsey, e.g.

```ruby
error&.body&.dig('messages')&.any? { |m| m['key'].include? INVALID_ADDRESS_ERROR }
```